src/Microsoft.DotNet.Darc/src/DarcLib/Actions/Remote.cs

```diff
@@ -465,6 +471,41 @@ private IEnumerable<DependencyDetail> CalculateLeavesOfCoherencyTrees(IEnumerabl
             return toUpdate;
         }
 
+        private async Task<DependencyGraphNode> BuildGraphAtDependency(
+            IRemoteFactory remoteFactory,
+            DependencyDetail rootDependency,
+            List<DependencyDetail> updateList,
+            Dictionary<string, DependencyGraphNode> nodeCache)
+        {
+            DependencyGraphBuildOptions dependencyGraphBuildOptions = new DependencyGraphBuildOptions()
+            {
+                IncludeToolset = true,
+                LookupBuilds = true,
+                NodeDiff = NodeDiff.None,
+                EarlyBuildBreak = new EarlyBreakOn
+                {
+                    Type = EarlyBreakOnType.Assets,
+                    BreakOn = new List<string>(updateList.Select(d => d.Name))
+                }
+            };
+
+            DependencyGraph dependencyGraph =
+                await DependencyGraph.BuildRemoteDependencyGraphAsync(remoteFactory,
+                    null, rootDependency.RepoUri, rootDependency.Commit,
+                    dependencyGraphBuildOptions, _logger);
+
+            // Cache all nodes in this built graph.
+            foreach (DependencyGraphNode node in dependencyGraph.Nodes)
+            {
+                if (!nodeCache.ContainsKey($"{node.RepoUri}@{node.Commit}"))
+                {
+                    nodeCache.Add($"{node.RepoUri}@{node.Commit}", node);
+                }
+            }
```

Comment:
I think this is mostly a nit. Above you build the dependency graph by calling `DependencyGraph.BuildRemoteDependencyGraph` which already iterates through the nodes which are returned. If you pass `nodeCache` you could build the cache on the fly and save this cycle here.